syntax = "proto3";

package solana.mev.control;

// Control command with Ed25519 signature support
message Command {
  string id = 1;
  string module = 2;
  string action = 3;
  map<string, string> params = 4;
  uint64 nonce = 5;          // Anti-replay nonce
  uint64 timestamp_ns = 6;
  bytes signature = 7;        // Ed25519 signature
  string pubkey_id = 8;       // Public key identifier
}

// Policy update command
message PolicyUpdate {
  string policy_id = 1;
  string policy_type = 2;
  map<string, double> thresholds = 3;
  map<string, string> rules = 4;
  bool enabled = 5;
  uint64 effective_from = 6;
}

// Model swap command
message ModelSwap {
  string model_id = 1;
  string model_path = 2;
  string model_type = 3;
  string version = 4;
  map<string, string> metadata = 5;
}

// Kill switch command
message KillSwitch {
  string target = 1;
  string reason = 2;
  uint64 duration_ms = 3;
  bool force = 4;
}

// Configuration update
message ConfigUpdate {
  string config_key = 1;
  string config_value = 2;
  string config_type = 3;
  bool hot_reload = 4;
}